### ISSUE TITLE
Cache

### ISSUE THREAD

**Issue by ray73864** [diladele/squid3-windows#40]
Having trouble using the cache set up, I have it set as

````
/cygdrive/c/Squid/var/cache/squid/
````

I then stop the squid service, go to the squid-bin directory in command prompt (as administrator) and type 'squid -z' then I start the squid service again.

The problem is that it continuosly likes to complain about the fact that it can't find the cachedir '00' and that I should run 'squid -z' even though I have multiple times.

**Comment by ea-at-diladele-com** [comment]
Hi ray73864,

did you precisely follow the steps described here?
http://docs.diladele.com/faq/squid/cache_squid_windows.html

Thank you!
-e

**Comment by ray73864** [comment]
I did, yes. Followed it exactly several times and not once did it work for
me. This is on Windows Server 2012 R2.

**Comment by RostorX** [comment]
cache_dir aufs c:/squidcachedir/cache 3000 16 256

This works on Win2008R2

**Comment by ray73864** [comment]
That's basically what I have on my Win2k12R2 server, but it isn't working
as it continously tells me in the logfile that I need to run 'squid -z'
even though I have several times, and when I look in my squid cachedir,
there isn't any directories in there that should have been created, such as
00, 01, etc...

**Comment by RostorX** [comment]
Did you give EVERYONE rights to cachedir folder ?

**Comment by ray73864** [comment]
I would have to check.

If that is what the issue is then I would respectfully ask that the wiki be
updated to reflect that. I made the assumption (possibly a faulty one at
that) that because I was logged in as the administrator and running 'squid
-z' as the administrator when I open up the command prompt (yes, I am doing
'Command Prompt (Admin)' when I right click on the start button) that it
would use the administrator privs to set up the cachedir.

**Comment by ea-at-diladele-com** [comment]
Hi ray,

Did you have a chance to check RostorX's suggestion? Did it solve you problem?
Thanks!
-e

**Comment by ray73864** [comment]
I won't be able to try it until later on today when I can get to the
primary school I work at to try it out.

Will come back with a yay or nay then.

**Comment by ea-at-diladele-com** [comment]
did it work? thank you!

**Comment by ra-at-diladele-com** [comment]
3.5.6 is on site - squid.diladele.com

**Comment by ea-at-diladele-com** [comment]
The symptoms are you are not running the "squid -z" from the squid terminal (should be on the desktop after installation), but a simple cmd.exe

If that is the case , please make sure C:\Squid\bin is in the %PATH% variable

**Comment by ea-at-diladele-com** [comment]
I would appreciate if you could confirm. Thank you!

**Comment by ray73864** [comment]
That appears to have worked on my test machine, though I won't be able to test it on site for a couple of weeks as the site (in this case a primary school) is locked up due to school holidays.

**Comment by ea-at-diladele-com** [comment]
Thank you! I will close this issue for now, if you still bump into any troubles drop an email at support @ diladele.com or open a new issue.

**Comment by mkraju** [comment]
Iam also facing same problem.
squid conf -- cache_dir ufs c:/squid/var/cache/squid/cache 100 16 256
when i run squid -z getting error "assertion failed: Kid.cc:45: "cpid > 0" in cache.log
hence its not creating swap folders in cache directory
gave full rights for everyone on cache folder

**Comment by ea-at-diladele-com** [comment]
Please make sure you precisely follow the steps described here
http://docs.diladele.com/faq/squid/cache_squid_windows.html

You have to perform "squid -z" in the **Squid Terminal window that the squid installer creates on your desktop.**

**Comment by mkraju** [comment]
YES, Its working now.
Thanks.

**Comment by nebadom** [comment]
I was having a lot of trouble configuring the cache: squid -z returned errors in the cache log. No matter how much I follow the steps to the letter, the squid -z was reluctant to initialize the cache folders.

Then I read about the %PATH%: So I added "C:\Squid\bin" to the %PATH% (please, for the love of Kthulu do not REPLACE the existing PATH value, just add the new string).

And then I attempted squid -z IN the cache folder and voilà! it worked!!!

So I would consider %PATH% > add > "C:\Squid\bin" to be the answer I was looking for.

**Comment by ra-at-diladele-com** [comment]
Ok clear. Thanks a lot for your research!

**Comment by DRSDavidSoft** [comment]
@ra-at-diladele-com Hopefully, the part where `Squid\bin` needs to be in `%PATH%` should be documented!  I usually don't use the shortcuts created, and just `cd` into the squid `bin` dir.  The `assertion failed: Kid.cc:45: "cpid > 0"` gave me a headache, where it just needed to be in path. 

#### tl;dr
- `cd C:\Squid\bin`
- `path %cd%;%path%`
- `squid.exe -z`

That's it :)

**Comment by hwaarl** [comment]
Also: please note that you must install squid onto an unencrypted drive/partition. Mine had Bitlocker activated and it took me hours to realise. Uninstall, re-install on a different drive with the same conf - fixed
